### PR TITLE
Module5

### PR DESCRIPTION
**Pull request by wishy78**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+FROM python:slim-buster as base
+# update Image
+RUN apt-get update
+# install poetry
+RUN pip install poetry
+# copy all except in dockerignore
+COPY . .
+# install prerequisits
+RUN poetry install --no-root --no-dev
+RUN pip install -r requirements.txt
```
Comment (BenRamchandani) [inline Dockerfile:11]: The poetry install should be enough, shouldn't need pip

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+FROM base as production
+# install gunicorn
+RUN pip install gunicorn
+# Cmd /entrypoint
+CMD ["gunicorn"  , "-b", "0.0.0.0:5000", "todo_app.app:create_app()"]
+
+# dev image
+FROM base as development
+# install flask
+RUN pip install flask
```
Comment (BenRamchandani) [inline Dockerfile:27]: Similarly you shouldn't need to explicitly install flask, poetry should already have done so

File: README.md [hunk 3]
```diff
@@ -90,4 +90,33 @@ Note: this is pulling from Module 3 currently so change the version to master in
...
+# docker build -f .\Dockerfile --tag todo-app .
+# docker run -p 8181:5000 --env-file .\.env todo-app
+
+# Run Docker Image
+Run the following in powershell teminal in VSCODE
+```powershell
+# For Production
+docker run -d -p 8181:5000 --env-file .\.env --mount type=bind,source="$(pwd)"/todo_app,target=/app/todo_app todo-app:dev
+# for Development
+docker run -d -p 8182:5000 --env-file .\.env --mount type=bind,source="$(pwd)"/todo_app,target=/app/todo_app todo-app:prod
```
Comment (BenRamchandani) [inline README.md:110]: These two are the wrong way around!

**Review by BenRamchandani — COMMENTED** [review]
Your Dockerfile looks good and the instructions work for me.

I've left a couple of optional comments, but the reason I can't approve it is you've got your Trello token committed in your git history. Please [revoke the token](https://support.atlassian.com/trello/docs/revoking-a-trello-token/), then submit again.

**Comment by wishy78** [comment]
I thought it had in a later commit or do I have to revoke that commit first?

Thanks
Jon

**Comment by BenRamchandani** [comment]
@wishy78 you've removed it in a later commit, but:
* It's still visible in the git history https://github.com/wishy78/DevOps-Course-Starter/commit/c4dbffe09e27c30e6b01225184ffeb36ee149b67#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R21
* Even if you were to overwrite the history and remove the commit that added it, it's good practice to assume any secret pushed to a repository is compromised, especially a public one

Please revoke the token itself on Trello and generate a new one for you to use going forward. There's no harm in leaving the old one in the history then.

File: Dockerfile (hunk 1 above)
Comment (wishy78) [reply Dockerfile:11]: i hashed out line 6 Run pip install poerty but then got an error on Run poetry install --no-root --no-dev

Comment (wishy78) [reply Dockerfile:11]: so i have left this as is

File: Dockerfile (hunk 2 above)
Comment (wishy78) [reply Dockerfile]: This was ok to build with Run pip install flask hashed out

File: Dockerfile (hunk 1 above)
Comment (BenRamchandani) [reply Dockerfile:11]: Ah sorry, you will still need to install poetry (line 6), but poetry should take care of the rest of the dependencies, so you shouldn't need line 11

**Comment by BenRamchandani** [comment]
@wishy78 great, thanks 🙂 If you resubmit on Aptem I'll approve it

**Comment by wishy78** [comment]
@BenRamchandani  Added as a note in aptem

**Comment by BenRamchandani** [comment]
@wishy78 can you submit it as a document instead? Aptem won't mark the module as approved for notes 🙁

**Comment by wishy78** [comment]
@BenRamchandani  it allows me to submit now so i think ive done it

**Comment by BenRamchandani** [comment]
Done, looks like it worked this time 🙂

**Comment by wishy78** [comment]
@BenRamchandani  can i merge then as it says requires review?

**Review by BenRamchandani — APPROVED** [review]
Yeah sure, I've pressed approve 🙂

**Pull request merged**